File: GameLibrary/src/cc/lib/probot/Direction.java
package cc.lib.probot;

public enum Direction {
    Right(1, 0),
    Down(0, 1),
    Left(-1, 0),
    Up(0, -1),
    ;

    Direction(int dx, int dy) {
        this.dx = dx;
        this.dy = dy;
    }

    public final int dx, dy;

    Direction turn(int amt) {
        return Direction.values()[(ordinal()+amt+values().length)%values().length];
    }
}

File: Probot/src/cc/lib/probot/Direction.kt
package cc.lib.probot

import cc.lib.annotation.Keep

@Keep
enum class Direction(val dx: Int, val dy: Int) {
	Right(1, 0),
	Down(0, 1),
	Left(-1, 0),
	Up(0, -1);

	fun turn(amt: Int): Direction {
		return values()[(ordinal + amt + values().size) % values().size]
	}
}